### DISCUSSION TITLE
Uploading SARIF files for pull requests

### DISCUSSION THREAD

**Discussion by chungyc** [community/community#54013] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

The [code-scanning/sarifs](https://docs.github.com/en/rest/code-scanning?apiVersion=2022-11-28#upload-an-analysis-as-sarif-data) is not able to upload SARIF files for a pull request without write permission for `security-events`.  This means a pull request from a public forked repository cannot have code scanning results uploaded, because the [maximum access](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) for `security-events` in this case is `read`.

However, [github/codeql-action/upload-sarif](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/uploading-a-sarif-file-to-github) does not need write permission to upload SARIF files for pull requests, presumably because whatever undocumented API it uses relaxes this restriction.  Would it be possible to have the code-scanning/sarifs API behave the same?

**Comment by chungyc** [comment]
Alternatively, would it be an option to document the other API that github/codeql-action/upload-sarif uses?

**Comment by chrisgavin** [comment, marked as answer]
Hi @chungyc, thanks for the feedback.

We did not initially implement support for this in the public API, as we expected most Actions workflows would make use of the `github/codeql-action/upload-sarif` action for doing the actual upload of SARIF file. I can see how it would be useful to be able to use the public SARIF upload API from within an Actions workflow running on a pull request from a fork though.

I've created an internal issue for our team to discuss whether we could update the permissions on the public API to be more similar to the internal CodeQL Action API. I can't make any promises about if/when this change will be made, but we'll discuss whether it could be added to the roadmap.

**Comment by chrisgavin** [comment]
Hi again @chungyc. I'm happy to report we've now updated the behavior for the public Code Scanning upload API so the permissions match that of the endpoint of `github/codeql-action/upload-sarif`. I hope this helps. :smiley:

**Reply by chungyc** [reply]
Thank you so much!
